Sources/Components/SuggestShippingView/SuggestShippingView.swift

```diff
@@ -0,0 +1,139 @@
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        return label
+    }()
+
+    private lazy var suggestShippingbutton: Button = {
+        let button = Button(style: .suggestShippingStyle)
+        button.contentEdgeInsets = UIEdgeInsets(
+            top: .spacingS + .spacingXS,
+            leading: .spacingXL,
+            bottom: .spacingS + .spacingXS,
+            trailing: .spacingXL + imageHorizontalInset
+        )
+        button.titleEdgeInsets = UIEdgeInsets(
+            top: 0,
+            leading: imageHorizontalInset,
+            bottom: 0,
+            trailing: -imageHorizontalInset
+        )
+        button.translatesAutoresizingMaskIntoConstraints = false
+        let shippingTruck = UIImage(named: .shippingTruck)
+        let darkShippingTruck = shippingTruck.withTintColor(.textAction)
+        shippingTruck.imageAsset?.register(darkShippingTruck, with: .init(userInterfaceStyle: .dark))
+        button.setImage(shippingTruck, for: .normal)
+        return button
+    }()
+
+    private lazy var showInfoButton = Button(style: .flat)
+    private lazy var loadingIndicator = LoadingIndicatorView(withAutoLayout: true)
+
+    private func setup() {
+        layer.borderWidth = 1
+        layer.borderColor = UIColor.sardine.cgColor
+        layer.cornerRadius = 3
+
+        addSubview(horizontalContainer)
+        horizontalContainer.fillInSuperview(margin: .spacingM)
+        textContainer.addArrangedSubviews([
+            title,
+            message
+        ])
+        containerView.addArrangedSubviews([
+            textContainer,
+            suggestShippingbutton
+        ])
+        horizontalContainer.addArrangedSubview(containerView)
+
+        NSLayoutConstraint.activate([
+            loadingIndicator.heightAnchor.constraint(equalToConstant: 30),
+            loadingIndicator.widthAnchor.constraint(equalToConstant: 30),
+        ])
+
+        viewModel
+            .$state
+            .sink { [weak self] newState in
+                self?.applyState(newState)
+            }
+            .store(in: &cancellables)
+    }
+
+    @objc
+    private func didTapButton() {
+        viewModel.suggestShipping()
+    }
+
+    private func applyState(_ newState: SuggestShippingViewModel.State) {
+        switch newState {
+        case .suggestShipping:
+            title.text = viewModel.title
+            message.text = viewModel.message
+            suggestShippingbutton.setTitle(viewModel.buttonTitle, for: .normal)
+            buttonCancellable = suggestShippingbutton
+                .publisher(for: .touchUpInside)
+                .sink { [weak self] _ in
+                    self?.viewModel.suggestShipping()
+                }
+
+        case .processing:
+            containerView.alpha = 0
+            addSubview(loadingIndicator)
+            loadingIndicator.centerInSuperview()
+            loadingIndicator.startAnimating()
+            layoutIfNeeded()
```

Comment:
The spinner won't be removed/stopped if it's started? Only by a state change that replaces the view on the object page?